controller/sharding/sharding.go

```diff
@@ -124,10 +124,20 @@ func LegacyDistributionFunction(replicas int) DistributionFunction {
 		} else {
 			h := fnv.New32a()
 			_, _ = h.Write([]byte(id))
-			shard := int32(h.Sum32() % uint32(replicas))
-			log.Debugf("Cluster with id=%s will be processed by shard %d", id, shard)
-			return int(shard)
+			shard := h.Sum32()
+
+			// Check if the shard value exceeds the maximum value for int32
+			if shard > uint32(math.MaxInt32) {
```

Comment:
But we'll take modulo replicas later. Should we only care if the number of replicas is somehow overflowing?